modules/adikteevBidAdapter.js

```diff
@@ -4,88 +4,208 @@ import * as utils from '../src/utils';
+    } catch (ex) {
+      canvas = undefined;
+      return false;
+    }
+  }
+
+  return !!gl;
+};
+
+export const getDeviceInfo = (deviceDimensions, viewDimensions, documentDimensions, webGL) => {
+  return {
+    browserWidth: viewDimensions.width,
+    browserHeight: viewDimensions.height,
+    deviceWidth: deviceDimensions.width,
+    deviceHeight: deviceDimensions.height,
+    documentWidth: documentDimensions.width,
+    documentHeight: documentDimensions.height,
+    webGL: webGL,
+  };
+};
+
+const validateSizes = sizes => utils.isArray(sizes) && sizes.some(size => utils.isArray(size) && size.length === 2);
 
 export const spec = {
   code: BIDDER_CODE,
   supportedMediaTypes: [BANNER],
 
   isBidRequestValid: (bid) => {
-    setstagingEnvironmentSwitch(stagingEnvironmentSwitch || !!bid.params.stagingEnvironment);
     return !!(
       bid &&
       bid.params &&
-      bid.params.bidFloorPrice &&
-      bid.params.placementId &&
+      bid.params.adSpaceId &&
       bid.bidder === BIDDER_CODE &&
       validateSizes(bid.mediaTypes.banner.sizes)
     );
   },
 
   buildRequests: (validBidRequests, bidderRequest) => {
-    const payload = {
-      validBidRequests,
-      bidderRequest,
-      refererInfo: bidderRequest.refererInfo,
-      currency: config.getConfig('currency'),
-      userAgent: navigator.userAgent,
-      screen: {
-        width: window.screen.width,
-        height: window.screen.height
+    const payload = Object.assign({},
+      {
+        akPbjsVersion: AK_PBJS_VERSION,
+        bidRequests: validBidRequests.map(conformBidRequest),
+        currency: config.getConfig('currency'),
+        debug: akDebug(utils.getParameterByName('akDebug'), config.getConfig('adikteev.debug')),
+        language: navigator.language,
+        refererInfo: bidderRequest.refererInfo,
+        deviceInfo: getDeviceInfo(getDeviceDimensions(), getViewDimensions(), getDocumentDimensions(), isWebGLEnabled()),
+        userAgent: navigator.userAgent,
       },
-      language: navigator.language,
-      cookies: document.cookie.split(';'),
-      prebidUpdateVersion: '1.29.0',
-    };
+      akOverrides(utils.getParameterByName('akOverrides'), config.getConfig('adikteev.overrides')));
+
     return {
       method: 'POST',
-      url: stagingEnvironmentSwitch ? ENDPOINT_URL_STAGING : ENDPOINT_URL,
+      url: endpointUrl(utils.getParameterByName('akEnv'), config.getConfig('adikteev.env')),
       data: JSON.stringify(payload),
     };
   },
 
-  interpretResponse: (serverResponse, bidRequests) => {
-    const returnedBids = [];
-    const validBidRequests = JSON.parse(bidRequests.data).validBidRequests;
-    serverResponse.body.forEach((value, index) => {
-      const overrides = {
-        requestId: validBidRequests[index].bidId,
-      };
-      returnedBids.push(Object.assign({}, value, overrides));
-    });
-    return returnedBids;
-  },
+  interpretResponse: (serverResponse) => serverResponse.body,
```

Comment:
The response that's coming back with the test parameters isn't valid json and is causing the bid to get dropped.

[adikteev_response.txt](https://github.com/prebid/Prebid.js/files/2880536/adikteev_response.txt)